events/codepipeline_cloudwatch.go

```diff
@@ -0,0 +1,106 @@
+	CodePipelineStageStateSucceeded CodePipelineStageState = "SUCCEEDED"
+	CodePipelineStageStateResumed   CodePipelineStageState = "RESUMED"
+	CodePipelineStageStateFailed    CodePipelineStageState = "FAILED"
+	CodePipelineStageStateCanceled  CodePipelineStageState = "CANCELED"
+)
+
+type CodePipelineState string
+
+const (
+	CodePipelineStateStarted    CodePipelineState = "STARTED"
+	CodePipelineStateSucceeded  CodePipelineState = "SUCCEEDED"
+	CodePipelineStateResumed    CodePipelineState = "RESUMED"
+	CodePipelineStateFailed     CodePipelineState = "FAILED"
+	CodePipelineStateCanceled   CodePipelineState = "CANCELED"
+	CodePipelineStateSuperseded CodePipelineState = "SUPERSEDED"
+)
+
+type CodePipelineActionState string
+
+const (
+	CodePipelineActionStateStarted   CodePipelineActionState = "STARTED"
+	CodePipelineActionStateSucceeded CodePipelineActionState = "SUCCEEDED"
+	CodePipelineActionStateFailed    CodePipelineActionState = "FAILED"
+	CodePipelineActionStateCanceled  CodePipelineActionState = "CANCELED"
+)
+
+// CodePipelineEvent is documented at:
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#codepipeline_event_type
+type CodePipelineCloudWatchEvent struct {
+	// Version is the version of the event's schema.
+	Version string `json:"version"`
+
+	// ID is the GUID of this event.
+	ID string `json:"id"`
+
+	// DetailType informs the schema of the Detail field. For deployment state-change
+	// events, the value should be equal to CodePipelineDeploymentEventDetailType.
+	// For instance state-change events, the value should be equal to
+	// CodePipelineInstanceEventDetailType.
+	DetailType string `json:"detail-type"`
+
+	// Source should be equal to CodePipelineEventSource.
+	Source string `json:"source"`
+
+	// AccountID is the id of the AWS account from which the event originated.
+	AccountID string `json:"account"`
+
+	// Time is the event's timestamp.
+	Time time.Time `json:"time"`
+
+	// Region is the AWS region from which the event originated.
+	Region string `json:"region"`
+
+	// Resources is a list of ARNs of CodePipeline applications and deployment
+	// groups that this event pertains to.
+	Resources []string `json:"resources"`
+
+	// Detail contains information specific to a deployment event.
+	Detail CodePipelineEventDetail `json:"detail"`
+}
+
+type CodePipelineEventDetail struct {
+	Pipeline string `json:"pipeline"`
+
+	// From live testing this is always int64 not string as documented
+	Version float64 `json:"version"`
+
+	ExecutionId string `json:"execution-id"`
+
+	Stage string `json:"stage"`
+
+	Action string `json:"action"`
+
+	State CodePipelineState `json:"state"`
+
+	Region string `json:"region"`
+
+	Type CodePipelineEventDetailType `json:"type"`
+}
+
+type CodePipelineEventDetailType struct {
+	Owner string `json:"owner"`
+
+	Category string `json:"category"`
+
+	Provider string `json:"provider"`
+
+	Version string `json:"version"`
```

Comment:
example from docs shows an int, not a string https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#codepipeline_event_type

Reply:
verified with the schema published to the EventBridge schema registry, this field should be change to an int

```
...
      "Type": {
        "type": "object",
        "required": ["owner", "provider", "category", "version"],
        "properties": {
          "owner": {
            "type": "string"
          },
          "provider": {
            "type": "string"
          },
          "category": {
            "type": "string"
          },
          "version": {
            "type": "number"
          }
        }
      },
...
```